rllib/algorithms/algorithm.py

```diff
@@ -1141,9 +1140,11 @@ def evaluate(
             eval_results = {}
 
         if self.config.enable_env_runner_and_connector_v2:
-            eval_results = self.metrics.reduce(
-                key=EVALUATION_RESULTS, return_stats_obj=False
-            )
+            eval_results = self.metrics.peek(key=EVALUATION_RESULTS, default={})
+            if not eval_results:
+                logger.warning(
+                    "No evaluation results found for this iteration. This can happen if the evaluation worker(s) is/are not healthy."
+                )
```

Comment:
This is a branch of code that we hit in eval worker failure tests, where eval results will not be part of the metrics.